core/domain/exp_domain.py

```diff
@@ -1343,8 +1348,7 @@ def validate(self, exp_param_specs_dict):
 
         if self.hints:
             if self.solution:
-                Solution.from_dict(
-                    self.id, self.solution).validate(self.id)
+                self.solution.validate(self.id)
```

Comment:
Right, I think it should still be in the interaction validate() method. What I meant was to split up the two things like this:

```
    self.solution.validate()

    if self.hints and not self.solution: (or if self.solution and not self.hints, or whatever it should be)
         error
```